### DISCUSSION TITLE
Show if a PlaylistTrack is a liked song or not, and set a song as a liked song

### DISCUSSION THREAD

**Discussion by KenthJ** [JohnnyCrazy/SpotifyAPI-NET#752] [Q&A] [answered]
Is it possible to get info on a PlaylistTrack if it´s a liked song or not (indicated with a heart in the original Spotify app)
And is there a method to set a specific song as a liked song

Thanks a lot in advance

**Comment by JohnnyCrazy** [comment, marked as answer]
Hi,

Checking if a song is liked is a separate API call:

https://developer.spotify.com/documentation/web-api/reference/#/operations/check-users-saved-tracks

Same for liking songs

https://developer.spotify.com/documentation/web-api/reference/#/operations/save-tracks-user

**Reply by KenthJ** [reply]
But isn't it possible to do it directly inside this SpotifyAPI-Net?

**Reply by JohnnyCrazy** [reply]
Sure, all endpoints are available and can be found via the name of the endpoint. In this case:

https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/7aea4e365ef49f2cb2ae2c00d567da0334af8857/SpotifyAPI.Web/Clients/Interfaces/ILibraryClient.cs#L60

https://github.com/JohnnyCrazy/SpotifyAPI-NET/blob/7aea4e365ef49f2cb2ae2c00d567da0334af8857/SpotifyAPI.Web/Clients/Interfaces/ILibraryClient.cs#L100

```c#
var checked = await spotify.Library.CheckTracks(new LibraryCheckTracksRequest(new List<string> { "mytrackid" }));

var response = await spotify.Library.SaveTracks(new LibrarySaveTracksRequest(new List<string> { "mytrackid" }));
```

**Reply by KenthJ** [reply]
Excellent, thanks a lot, just what I needed
